src/briefcase/platforms/linux/system.py

```diff
@@ -573,9 +581,13 @@ def output_format_template_context(self, app: AppConfig):
         # Add the vendor base
         context["vendor_base"] = app.target_vendor_base
 
-        # Use the non-root user if Docker is not mapping usernames
+        # Use the non-root user if Docker is not mapping usernames. Also use a non-root
+        # user if we're on macOS; user mapping doesn't alter Docker operation, but some
+        # packaging tools (e.g., Arch's makepkg) don't like running as root.
         try:
-            context["use_non_root_user"] = not self.tools.docker.is_user_mapped
+            context["use_non_root_user"] = self.use_docker and (
+                self.tools.host_os == "Darwin" or not self.tools.docker.is_user_mapped
+            )
         except AttributeError:
             pass  # ignore if not using Docker
```

Comment:
If we check `self.use_docker` here, then `use_non_root_user` will be `False` if rolling out a template without using Docker. While I suppose this setting is arbitrary in this case, this will reverse the current default of `True` from the templates.

It also effectively mitigates the need for the `try/except AttributeError` since `self.tools.docker` will exist if Docker is being used.

I'm not sure this setting is especially important....but throwing this out there.

Reply:
That's a good point - the template default is "use root user", so it's safer to fall back to that than set it to false because there's no Docker. I'll tweak the logic.

(Of course the real issue here is that briefcase create is sensitive to whether you used Docker, but that's a *much* bigger can of worms)